fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/TransactionObject.java

```diff
@@ -121,6 +123,25 @@ public List<PublicKey> getReceiversTransaction(Map<String, PublicKey> mapHashKey
     return receivers;
   }
 
+  /**
+   * Function that give the Public Key receiver and the amount on the Output of transaction
+   *
+   * @param mapHashKey Map<String,PublicKey> dictionary public key by public key hash
+   * @return List<PublicKey,Long> outputs public keys
+   */
+  public Map<PublicKey, Long> getReceiversTransactionMap(Map<String, PublicKey> mapHashKey) {
+    Iterator<String> receiverHashIte = getReceiversHashTransaction().iterator();
+    Map<PublicKey, Long> receivers = new HashMap<>();
+    while (receiverHashIte.hasNext()) {
+      PublicKey pub = mapHashKey.getOrDefault(receiverHashIte.next(), null);
+      if (pub == null) {
+        throw new IllegalArgumentException("The hash correspond to no key in the dictionary");
+      }
```

Comment:
First of, Map.get() of non-contained keys already returns null. So the getOrDefault is not necessary.

But anyway, I would recommend something more meaningful like :

```suggestion
       if (!mapHashKey.containsKey(receiverHashIte.next())) {
        throw new IllegalArgumentException("The hash correspond to no key in the dictionary");
      }
      
      PublicKey pub = mapHashKey.get(receiverHashIte.next());
```